web/pages/group/[...slugs]/index.tsx

```diff
@@ -196,85 +198,118 @@ export default function GroupPage(props: {
     </Col>
   )
 
-  const questionsTab = (
-    <ContractSearch
-      user={user}
-      defaultSort={'newest'}
-      defaultFilter={suggestedFilter}
-      additionalFilter={{ groupSlug: group.slug }}
-      persistPrefix={`group-${group.slug}`}
-    />
+  const questionsPage = (
+    <>
+      {/* align the divs to the right */}
+      <div className={' flex justify-end px-2 pb-2 sm:hidden'}>
+        <div>
+          <JoinGroupButton group={group} user={user} />
+        </div>
+      </div>
+      <ContractSearch
+        user={user}
+        defaultSort={'newest'}
+        defaultFilter={suggestedFilter}
+        additionalFilter={{ groupSlug: group.slug }}
+        persistPrefix={`group-${group.slug}`}
+      />
+    </>
   )
 
-  const tabs = [
+  const sidebarPages = [
     {
       badge: `${contractsCount}`,
       title: 'Markets',
-      content: questionsTab,
+      content: questionsPage,
       href: groupPath(group.slug, 'markets'),
+      key: 'markets',
     },
     {
       title: 'Leaderboards',
-      content: leaderboard,
+      content: leaderboardPage,
       href: groupPath(group.slug, 'leaderboards'),
+      key: 'leaderboards',
     },
     {
       title: 'About',
-      content: aboutTab,
+      content: aboutPage,
       href: groupPath(group.slug, 'about'),
+      key: 'about',
     },
   ]
 
-  const tabIndex = tabs
-    .map((t) => t.title.toLowerCase())
-    .indexOf(page ?? 'markets')
+  const pageContent = sidebarPages[sidebarIndex].content
+  const onSidebarClick = (key: string) => {
+    const index = sidebarPages.findIndex((t) => t.key === key)
+    setSidebarIndex(index)
+  }
+
+  const joinOrAddQuestionsButton = (
+    <JoinOrAddQuestionsButtons
+      group={group}
+      user={user}
+      isMember={!!isMember}
+    />
+  )
 
   return (
-    <Page>
-      <SEO
-        title={group.name}
-        description={`Created by ${creator.name}. ${group.about}`}
-        url={groupPath(group.slug)}
-      />
-      <Col className="relative px-3">
-        <Row className={'items-center justify-between gap-4'}>
-          <div className={'sm:mb-1'}>
-            <div
-              className={'line-clamp-1 my-2 text-2xl text-indigo-700 sm:my-3'}
-            >
-              {group.name}
-            </div>
-            <div className={'hidden sm:block'}>
-              <Linkify text={group.about} />
-            </div>
+    <>
+      <header className="sticky top-0 z-50 w-full pb-2 md:hidden lg:col-span-12">
```

Comment:
https://user-images.githubusercontent.com/6242616/190210177-4cb11d60-8792-4af9-be01-35af7c234f69.mov

Does anybody have a good idea for how to fix this? Tailwinds 'sticky' position uses the browser window as its position reference, so ContractsSearchFilters overlaps the header. What I want is the same behaviour, but with the component stopping at the header instead.

Reply:
Hmm, good question. I'm not sure. There might be a way to add an offset to the search bar's sticky y-offset.

Alternatively, you could just pass an option to disable the sticky search bar for groups on mobile.

Reply:
One small suggestion might be to extract this header to a new component in this file.